### DISCUSSION TITLE
Open a .tex file via macro

### DISCUSSION THREAD

**Discussion by Tealk** [texstudio-org/texstudio#1403] [Q&A] [answered]
Hi,

I would like to create one or more macros that open .tex files from a certain folder. It would be elegant if a selection would be shown to me, but it would also be enough for me if I could enter the 3-4 files there.

**Comment by sunderme** [comment]
have you looked into the manual ?
Chapter 4.6.3 Script Macros : 
`readFile(name) ` , `app.fileOpen/Save/Close/.../editUndo/.../QuickBuild/... ` , `fileChooser.exec() `?

**Reply by Tealk** [reply]
But unfortunately I have no idea how to use the whole thing.
I would like to close the current file and open another one.

My try:
```
%SCRIPT
app.close
app.load("/Users/tealk/GIT/butterfly-aspect/Grundregelwerk.tex")
```

**Comment by sunderme** [comment, marked as answer]
```
%SCRIPT
app.fileClose()
app.load("/filename here")
```
works ...

**Reply by Tealk** [reply]
is there also a command to close all open tabs?

**Reply by sunderme** [reply]
```
app.fileCloseAll()
```

**Comment by sunderme** [comment]
`app.fileOpen()` asks for the file name ...

**Reply by Tealk** [reply]
yes but since there are so many files in the folder, this is not nice to select so I prefer to do this then via individual macros for each file
